tf2/include/tf2/LinearMath/Matrix3x3.hpp

```diff
@@ -296,16 +296,24 @@ class Matrix3x3 {
 		Euler euler_out2; //second solution
 		//get the pointer to the raw data
 
-		// Check that pitch is not at a singularity
-  		// Check that pitch is not at a singularity
-		if (tf2Fabs(m_el[2].x()) >= 1)
+		// Apply epsilon thresholding to matrix elements to handle numerical precision issues
+		// Use a conservative threshold to handle numerical errors from quaternion-to-matrix conversion
+		tf2Scalar threshold = tf2Scalar(1e-8);
```

Comment:
Curious what the motivation for using 1e-8 here is?  It's smaller than FLT_EPSILON and bigger than DBL_EPSILON